esp-wifi/src/compat/misc.rs

```diff
@@ -20,8 +21,26 @@ unsafe extern "C" fn fclose(stream: *const ()) -> i32 {
     todo!("fclose {:?}", stream);
 }
 
-// not available in ROM on ESP32-S2
-#[cfg(feature = "esp32s2")]
+// We cannot just use the ROM function since it needs to allocate memory
+#[no_mangle]
+unsafe extern "C" fn strdup(str: *const i8) -> *const u8 {
```

Comment:
Given the recent pains I'd prefer if this was a `*const c_char -> *const c_char` function.